### ISSUE TITLE
Development mode prevents errors going into console

### ISSUE THREAD

**Issue by nojacko** [damiencaselli/ember-cli-sentry#16]
When development is set to true all errors are completely hidden. Is there away to have them show in the developer console like they would if this plugin wasn't installed? Thanks

**Comment by damiencaselli** [comment]
Hi @nojacko 

This is actually a bit tricky since `ember-cli-sentry` uses [ravenjs ember plugin](https://github.com/getsentry/raven-js/blob/master/plugins/ember.js), which captures the exception.

Way back then it was not the case, we changed it in 1b8b756471cb478d312a20bf135cf81a324065ae by @rwjblue.

This change is actually quite good since it allows to benefit from plugin changes, but this is actually quite questionnable since it overwrites `Ember.onerror` and really messes up with logs with no clean possibility to keep standard error catching.

To answer your question: what you want is redefine the `Ember.onerror` in an initializer of your own that would call the already defined `Ember.onerror` (which is what the raven.js Ember plugin does).

```javascript
// /instance-initializers/wheresmyerror.js

export function initialize(application) {

  let _onerror = Ember.onerror;

  Ember.onerror = function(error) {
    if (typeof _onerror === 'function') {
      _onerror.call(this, error);
    }
    // error handling, for example
    // throw error;
  }

  Ember.RSVP.on('error', function(error) {
    if (error !== undefined) {
      // error handling
    }
  });
}

export default {
  initialize,
  name: 'wheresmyerror'
};
```

Typing this makes little to no sense at all, but this would work I guess.

I'm wondering is whether or not`ember-cli-sentry` should use the Ember plugin or not.

Any thought about it @rwjblue? I tried to ping @mattrobenolt about it with no luck so far.

**Comment by mattrobenolt** [comment]
What's up?

**Comment by damiencaselli** [comment]
Hey @nojacko, I'm planning on doing rather heavy changes on the way this addon works.
I've pushed a branch that _should_ work here: https://github.com/damiencaselli/ember-cli-sentry/tree/tsugi

It solves the problem you encounter by proxying the Raven object.

Could you please give it a shot and tell me what could be improved?

@mattrobenolt thanks for stopping by :bow: 

Since Emberjs now encourages the use of _ember-cli_, do you think the emberjs plugin version of ravenjs should follow this tendency?

Doing this would avoid such problems as this one and [that one](https://github.com/damiencaselli/ember-cli-sentry/issues/17) by letting users choose how they want to handle `Ember.onerror` and `RSVP` errors.

Of course one could just install the _emberless_ version of the plugin and handle it by themselves, but being backed by the GetSentry team would be awesome.

I've been working on a major refactor of this addon (see [here](https://github.com/damiencaselli/ember-cli-sentry/tree/tsugi)) that addresses those issues, could you please have a look at it?
I'd gladly have someone from the GetSentry team to help maintain this repository, or give some advices|directions.

Cheers!

**Comment by mattrobenolt** [comment]
> Since Emberjs now encourages the use of ember-cli, do you think the emberjs plugin version of ravenjs should follow this tendency?

Yeah, absolutely. I'm not an Ember developer, so this stuff is a bit foreign to me. We're down for supporting whatever the community needs.

> I'd gladly have someone from the GetSentry team to help maintain this repository, or give some advices|directions.

I'd be happy to provide some general direction, but again, I'm not an ember user/developer, so my context is limited. I'm happy to support whatever makes sense.

**Comment by damiencaselli** [comment]
Great!

So I guess moving the official emberjs plugin logic part to this add-on would be a good addition, so people can override default behaviour without much effort.

**Comment by devinus** [comment]
@damiencaselli When can we expect the new branch to drop? Excited about it.

**Comment by damiencaselli** [comment]
I'm betting on a release this weekend.
The branch is actually pretty stable, but documentation has to be updated (who said "done"?), and I need to integrate the non-ember-cli plugin code to replicate current working state.

**Comment by damiencaselli** [comment]
… or later this week more likely :smile_cat:

**Comment by damiencaselli** [comment]
This should be fixed in the latest release (2.0.0-beta.1), feel free to let me know if you find any bug.

**Comment by damiencaselli** [comment]
@devinus did you have the chance to test add-on next version beta (>= 2.0.0-beta.1)?

**Comment by devinus** [comment]
@damiencaselli Not yet, sorry. We had to use the old version and I'm looking for an opportunity to revisit it again.

**Comment by damiencaselli** [comment]
I consider this issue resolved when updating to latest 2.0.0 (beta or not).
